ui/src/views/ResourcePreview.vue

```diff
@@ -65,7 +65,7 @@ export default defineComponent({
 
     this.resource = resource
     this.properties = [...resourcePredicates].map((predicate) => {
-      const values = resource.out(predicate).map((pointer: GraphPointer) => {
+      const values: (Term | RdfResource)[] = resource.out(predicate).map((pointer: GraphPointer) => {
```

Comment:
Why was this made explicit? What was the inferred type?

Reply:
```
ERROR in src/views/ResourcePreview.vue:67:5
TS2322: Type '[Quad_Predicate, (BaseQuad | DefaultGraph | BlankNode | Variable | Literal | (RdfResource<DatasetCore<Quad, Quad>> & ResourceIndexer<...>))[]][]' is not assignable to type '[Term, (Term | RdfResourceImpl<DatasetCore<Quad, Quad>>)[]][]'.
  Type '[Quad_Predicate, (BaseQuad | DefaultGraph | BlankNode | Variable | Literal | (RdfResource<DatasetCore<Quad, Quad>> & ResourceIndexer<...>))[]]' is not assignable to type '[Term, (Term | RdfResourceImpl<DatasetCore<Quad, Quad>>)[]]'.
    Type at position 1 in source is not compatible with type at position 1 in target.
      Type '(BaseQuad | DefaultGraph | BlankNode | Variable | Literal | (RdfResource<DatasetCore<Quad, Quad>> & ResourceIndexer<...>))[]' is not assignable to type '(Term | RdfResourceImpl<DatasetCore<Quad, Quad>>)[]'.
        Type 'BaseQuad | DefaultGraph | BlankNode | Variable | Literal | (RdfResource<DatasetCore<Quad, Quad>> & ResourceIndexer<...>)' is not assignable to type 'Term | RdfResourceImpl<DatasetCore<Quad, Quad>>'.
          Type 'RdfResource<DatasetCore<Quad, Quad>> & ResourceIndexer<RdfResource<DatasetCore<Quad, Quad>>>' is not assignable to type 'Term | RdfResourceImpl<DatasetCore<Quad, Quad>>'.
            Type 'RdfResource<DatasetCore<Quad, Quad>> & ResourceIndexer<RdfResource<DatasetCore<Quad, Quad>>>' is missing the following properties from type 'RdfResourceImpl<DatasetCore<Quad, Quad>>': __initialized, __initializeProperties
    65 |
    66 |     this.resource = resource
  > 67 |     this.properties = [...resourcePredicates].map((predicate) => {
       |     ^^^^^^^^^^^^^^^
    68 |       const values = resource.out(predicate).map((pointer: GraphPointer) => {
    69 |         if (pointer.term.termType === 'NamedNode') {
    70 |           return RdfResource.factory.createEntity(pointer)
```